lib/shared/addon/mixins/reservation-check.js

```diff
@@ -197,4 +197,26 @@ export default Mixin.create({
 
     return errors
   },
+
+  updatePreRequest() {
```

Comment:
Have you considered refactoring this function to take input and return output rather than modify data directly on the consumer of the mixin? Or perhaps giving this method a more verbose name like updateCpuMemoryPreRequest? Mixin's can be somewhat nebulous and it is not very clear what this function does from both places it's called.